internal/provider/storage/profile_datasource.go

```diff
@@ -0,0 +1,257 @@
+	Default             types.Bool   `tfsdk:"default"`
+	Enabled             types.Bool   `tfsdk:"enabled"`
+	IopsAllocated       types.Int64  `tfsdk:"iops_allocated"`
+	Units               types.String `tfsdk:"units"`
+	IopsLimitingEnabled types.Bool   `tfsdk:"iops_limiting_enabled"`
+	MaximumDiskIops     types.Int64  `tfsdk:"maximum_disk_iops"`
+	DefaultDiskIops     types.Int64  `tfsdk:"default_disk_iops"`
+	DiskIopsPerGbMax    types.Int64  `tfsdk:"disk_iops_per_gb_max"`
+	IopsLimit           types.Int64  `tfsdk:"iops_limit"`
+}
+
+// Init Initializes the data source.
+func (d *profileDataSource) Init(ctx context.Context, dm *profileDataSourceModel) (diags diag.Diagnostics) {
+	d.vdc, diags = vdc.Init(d.client, dm.VDC)
+	if diags.HasError() {
+		return
+	}
+
+	return
+}
+
+func (d *profileDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + categoryName + "_profile"
+}
+
+func (d *profileDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = d.superSchema(ctx).GetDataSource(ctx)
+}
+
+func (d *profileDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.CloudAvenue)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *client.CloudAvenue, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	d.client = client
+}
+
+func (d *profileDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	config := &profileDataSourceModel{}
+
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, config)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Init the resource
+	resp.Diagnostics.Append(d.Init(ctx, config)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	/*
+		Implement the data source read logic here.
+	*/
+
+	storageProfileID, err := d.vdc.FindStorageProfileID(config.Name.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Storage profile not found",
+			fmt.Sprintf("Storage profile %s not found in VDC %s", config.Name.ValueString(), d.vdc.GetName()),
+		)
+		return
+	}
+
+	storageProfileRef, err := d.vdc.GetStorageProfileReference(storageProfileID, true)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Storage profile not found",
+			fmt.Sprintf("Storage profile %s not found in VDC %s", config.Name.ValueString(), d.vdc.GetName()),
+		)
+		return
+	}
+
+	storageProfile, err := d.client.Vmware.GetStorageProfileByHref(storageProfileRef.HREF)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Storage profile not found",
```

Comment:
Change error message to defer with the precedent: "Storage Profile (Href) not found"